pkg/daemon/ceph/osd/volume.go

```diff
@@ -956,17 +956,19 @@ func GetCephVolumeRawOSDs(context *clusterd.Context, clusterInfo *client.Cluster
 				}
 			}
 			if encryptedBlock == "" {
-				// The encrypted block is not opened, this is an extreme corner case
-				// The OSD deployment has been removed manually AND the node rebooted
-				// So we need to re-open the block to re-hydrate the OSDInfo.
-				//
-				// Handling this case would mean, writing the encryption key on a temporary file, then call
-				// luksOpen to open the encrypted block and then call ceph-volume to list against the opened
-				// encrypted block.
-				// We don't implement this, yet and return an error.
-				return nil, errors.Errorf("failed to find the encrypted block device for %q, not opened?", block)
+				// The encrypted block is not opened.
+				// The OSD deployment has been removed manually.
+				// Let's re-open the block to re-hydrate the OSDInfo.
+				logger.Debugf("encrypted block device %q is not open, opening it now", block)
+				passphrase := os.Getenv(oposd.CephVolumeEncryptedKeyEnvVarName)
+				target := fmt.Sprintf("%s-%s", os.Getenv(oposd.PVCNameEnvVarName), oposd.DmcryptBlockType)
```

Comment:
I have concerns about this. I'm not sure how strictly guaranteed the behavior we are relying on is. I wonder if there are kubernetes versions or variants where the pvc name is not part of the dm name.

I think we should be able to use the `encryptedBlock` variable here, no? That should contain `-dmcrypt` according to the lines above.

Reply:
> I have concerns about this. I'm not sure how strictly guaranteed the behavior we are relying on is. I wonder if there are kubernetes versions or variants where the pvc name is not part of the dm name.
> 

This target name can be anything since we'll be opening luks with this target explicitly and call c-v on this.

> I think we should be able to use the `encryptedBlock` variable here, no? That should contain `-dmcrypt` according to the lines above.

c-v prefers to have /dev/mapper/<name> format, that's why i kept both the variables separate.

Reply:
I see. `target` is what `luksOpen` uses for its `name` parameter and is created by opening the disk. Thanks :)